docs/characterization/parallel_xeb.ipynb

```diff
@@ -0,0 +1,267 @@
+{
```

Comment:
<p>This is beyond this PR but if instead of simulating circuits inside of this method, you would move it to the very beginning where circuits are actually generated then in the future you could serialize circuits and simulation results together, potentially making the fidelity estimation very fast since simulation results can be done ahead of execution on the hardware.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/3760/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
<p>The simulation is rather fast (&lt; 1 second). The fact that we use one circuit library instead of a new random circuit for each pair makes this scale much better for large devices.</p><p> </p><p>For benchmarking, this sort of thing could be worth it (where you pre-generate one library of circuits and you pre-simulate all of them). But for characterization, this simulation needs to be done in a hot loop (i.e. once per optimization step with the new parameters) so you can't amortize the simulation cost. I do want to see how hard it would be to throw these two-qubit circuits into a symbolic math library and get the final probabilities in terms of the parameters. The other option is to slice into the qsim API which should be very fast for these two qubit circuits. </p>